MetaMask/metamask-extension
# Improve transaction confirmation page performance

## Explanation

This issue fixes a Flask issue by fixing an underlying problem with the selectors used in the transaction confirmation screen (also in stable). The issue arose in Flask because we relied on the `txData` object for a `useEffect`. The `txData` object was continuously updated through spreads etc without memoization. This PR fixes the problem by wrapping the existing logic in `reselect` selectors and enabling deep equal memoization for these selectors.

## More Information

Fixes https://github.com/MetaMask/snaps-monorepo/issues/828

## Manual Testing Steps

1. Create an transaction insight snap that logs every time it is called
2. View the transaction insight tab in the confirmation screen
3. See that the transaction insight snap only logs once

(Perhaps another way to test in stable?)
